src/fairseq2/recipes/wav2vec2/batch_weighted_datareader.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
this's an independant work we can skip it or release as indepedent PR

Reply:
We need to add this if we want the parquet datareaders to be part of the open-source recipe. Otherwise its not really worthwhile to show two recipes that independently work with a single dataloader.